lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -96,35 +112,28 @@ def run(self) -> bool:
                     self._install_if_needed(plugin),
                     self._configure_if_needed(plugin),
                     self._disable_if_needed(plugin),
+                    self._remove_if_needed(plugin),
```

Comment:
Since `disable()` above  does nothing else but deleting configuration... Ideally it should come after the plugin is removed.

However since we're removing the plugin
 1. when there's no more COS relation (i.e. no conusmer)
 2. right before we are actually removing the plugin
This should be safe enough (to what I understand).

In a worst case scenario we may get complaints in the logs.

I actually tried to go for a "post"-cleanup procedure... But with the current plugin lifecycle and status determination logic, it was virtually impossible to fit it into the scheme.

Reply:
Just for clarity: Of course not the `cleanup()` function itself. But to be able to reach that stage of the execution ;-)
(As in this case there are no grounds to get into a `DISABLED` state...)